### PR TITLE
🔨(frontend) encapsulate title component

### LINKED ISSUE #406: Make toolbar / waffle configurable
## Feature Request

openDesk wants to have a universal Header to access any openDesk applications and Account settings from. 

Currently all applications integrated in openDesk need to implement it individually.

It should look like this:
image

The specification of what functionality needs to be implemented is currently beeing discussed and this issue will be updated later.

Anything that could benefit other users and contribute to the general whitelabeling of Docs, should go into the `main` branch.
All customization in regards to openDesk should go in the `main-opendesk` branch for now.
Maybe a Fork with openDesk customizations would make sense in the future.

### PR DESCRIPTION
**Pull request by rvveber**
In order to modularize and simplify customizations to components, 
the title component is encapsulated.

closes #406

### CODE REVIEW

File: src/frontend/apps/impress/src/features/header/components/Title/Title.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      >
+        BETA
+      </Text>
+      <Text
+        $margin="none"
+        as="h2"
+        $color="#000091"
+        $zIndex={1}
+        $size="1.30rem"
+        $css="font-family: 'Marianne'"
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/header/components/Title/Title.tsx:30]: We don't need it actually.
```suggestion
```

File: src/frontend/apps/impress/src/features/header/components/Title/Title.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        $padding="2px 3px"
+        $size="8px"
+        $background="#368bd6"
+        $color="white"
+        $position="absolute"
+        $radius="5px"
+        $css={`
+                  bottom: 13px;
+                  right: -17px;
+                `}
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/header/components/Title/Title.tsx:20]:
```suggestion
        $css={`
          bottom: 13px;
          right: -17px;
        `}
```

**Review by AntoLC — COMMENTED** [review]
Very small changes.

I don't understand how this PR can close the issue #406, could you explain ?

**Comment by rvveber** [comment]
> Very small changes.
> 
> I don't understand how this PR can close the issue #406, could you explain ?

I originally created #406 due to the openDesks integration requiring a "central menu" in the header. 
(The issue's title was later updated to "make toolbar customizable.") 

However, since the requested change was too specific for this project and we aim to keep it generic, a more sophisticated solution would have been needed—something along the lines of: [matrix-org/matrix-react-sdk-module-api](https://github.com/matrix-org/matrix-react-sdk-module-api).

Unfortunately, I lacked the time and experience to implement such a large-scale task, which would also require extensive testing and user documentation.

Instead, i decided to override a single component (a single file) at build time, independent of this project. To maximize compatibility with future versions of Docs, I chose to override a very simple component that was already positioned where needed, which led to encapsulating the Title to its own file.

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,7 @@ and this project adheres to
 - ♻️(frontend) simplify stores #402
 - ✨(frontend) update $css Box props type to add styled components RuleSet #423
 - ✅(CI) trivy continue on error #453
+- 🔨(frontend) encapsulated title to its own component #474
```
Comment (AntoLC) [inline CHANGELOG.md:29]: When you will rebase main, you will see we have a new version, you will have to move this part under unrelease.

**Review by AntoLC — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (rvveber) [reply CHANGELOG.md]: Done

**Pull request merged**